tests/stack_test.go

```diff
@@ -811,3 +811,51 @@ func TestStackTags(t *testing.T) {
 	tags = lsTags()
 	assert.Equal(t, "projectValue", tags["projectTag"], "projectTag should be set to projectValue")
 }
+
+// Creates a unix timestamp string that can be used to avoid name collisions with existing stacks.
+// Takes an optional prefix string that will be joined with the timestamp string with a hyphen.
+func newUnixTimestampString(prefix string) string {
+	unixTsNanos := time.Now().UnixNano()
+	numericProjectName := strconv.Itoa(int(unixTsNanos))
+	if prefix == "" {
+		return numericProjectName
+	}
+	return fmt.Sprintf("%s-%s", prefix, numericProjectName)
+}
+
+//nolint:paralleltest // pulumi new is not parallel safe
+func TestNewStackConflictingOrg(t *testing.T) {
+	t.Parallel()
+
+	// This test requires the service, as only the service supports orgs.
+	if os.Getenv("PULUMI_ACCESS_TOKEN") == "" {
+		t.Skipf("Skipping: PULUMI_ACCESS_TOKEN is not set")
+	}
+
+	e := ptesting.NewEnvironment(t)
+	defer deleteIfNotFailed(e)
+
+	project := newUnixTimestampString("project")
```

Comment:
We use `resource.NewUniqueHex("test-name-", 8, -1)` in a few other places. We should probably just add a method to the environment object to do this and use that consistently everywhere.